File: stack/0682_baseball_game/0682_baseball_game.go
package main

import (
	"fmt"
	"strconv"
)

func calPoints(ops []string) int {
	stack := make([]int, 0)
	for _, s := range ops {
		if s == "C" {
			stack = stack[:len(stack)-1]
		} else if s == "D" {
			stack = append(stack, stack[len(stack)-1]*2)
		} else if s == "+" {
			stack = append(stack, stack[len(stack)-1]+stack[len(stack)-2])
		} else {
			temp, _ := strconv.Atoi(s)
			stack = append(stack, temp)
		}
	}

	res := 0
	for _, v := range stack {
		res += v
	}

	return res

}

func main() {
	ops := []string{"5", "2", "C", "D", "+"}
	res := calPoints(ops)
	fmt.Println(res)
}

File: stack/0682_baseball_game/0682_baseball_game.py
from typing import List


class Solution:
    def calPoints(self, ops: List[str]) -> int:
        res = []
        for op in ops:
            if op == 'C':
                res.pop()
            elif op == '+':
                res.append(res[-1] + res[-2])
            elif op == 'D':
                res.append(res[-1] * 2)
            else:
                res.append(int(op))
        return sum(res)


ops = ["5","2","C","D","+"]
res = Solution().calPoints(ops)
print(res)